src/maplibre_helpers.ts

```diff
@@ -229,18 +229,26 @@ export function setPrecision(pt: Position): Position {
 //  Helper for https://maplibre.org/maplibre-style-spec/expressions/#match.
 //  Gets one feature property, uses a map to match a key to a value, and
 //  includes a fallback if no keys match.
-export function matchValue<Output>(
+export function getDataDrivenPropertyValueSpecification<OutputType>(
   getter: any[],
-  map: { [name: string]: Output },
-  fallback: Output
-): DataDrivenPropertyValueSpecification<Output> {
+  map: { [name: string]: OutputType },
+  fallback: OutputType
+): DataDrivenPropertyValueSpecification<OutputType> {
+  // This looks weird because it is weird. I hate to add a comment but here we are.
```

Comment:
This is very verbose, do we want to check it in? Are the linked docs at https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/#match confusing? The interface of the function is hopefully clear, or we can change param names to make it better. The implementation is hopefully clear with a glance at the docs

Reply:
I am going to blame it on my addled illness brain but I didn't see those docs. Those docs seem fine